### ISSUE TITLE
[Feature] Expose and document `padding_mode: fill` and the corresponding `fill_value` in `K.RandomAffine`

### ISSUE THREAD

**Issue by MathiasBaumgartinger** [kornia/kornia#3090]
## 🚀 Feature
Currently, a call to `K.RandomAffine()` does not document and expose the `padding_mode: fill`. Additionally, the `fill_value` cannot be altered. 

## Motivation

I came across this problem, when working with semantic segmentation in torchgeo. The mask values, when augmented with a `K.RandomAffine()`, were shifted by -1 as discussed in https://github.com/microsoft/torchgeo/discussions/985#discussioncomment-11532847. This lead to very weird optimization problems which were not apparent first hand.

![image](https://github.com/user-attachments/assets/661f5f5f-831f-4f50-844f-3fdd973f6071)

Hence, I further propose an automatic application of the `padding_mode: fill` when working with masks.

## Pitch
This:
```py
from typing import Any, Dict, Optional
import torch
from torch import Tensor
import kornia.augmentation as K
from kornia.geometry.transform import warp_affine


class RandomAffineFill(K.RandomAffine):
    """
    Custom RandomAffine that fixes the output size to be the same as the input size.
    """

    def __init__(self, fill_value: int, *args: Any, **kwargs: Any) -> None:
        super().__init__(*args, **kwargs)
        self.fill_value = fill_value


    def apply_transform(
        self, input: Tensor, params: Dict[str, Tensor], flags: Dict[str, Any], transform: Optional[Tensor] = None
    ) -> Tensor:
        _, _, height, width = input.shape
        if not isinstance(transform, Tensor):
            raise TypeError(f"Expected the `transform` be a Tensor. Got {type(transform)}.")

        return warp_affine(
            input,
            transform[:, :2, :],
            (height, width),
            flags["resample"].name.lower(),
            align_corners=flags["align_corners"],
            padding_mode="fill",
            fill_value=torch.tensor([self.fill_value])
        )
```

And an automatic application on mask data.

**Comment by edgarriba** [comment]
@MathiasBaumgartinger are you open to send a PR updating the docs and interfaces ?

**Comment by MathiasBaumgartinger** [comment]
Might take a while but I added it to my TODOs :smile:

**Comment by nilaysrivastava** [comment]
Hi @edgarriba ,

I would like to work on this.

I recently merged a fix for affine broadcasting logic in #3388, so I am already familiar with the affwarp.py and Augmentation internals.

My plan is to:

- Update RandomAffine API: Modify `__init__` to accept `padding_mode` and `fill_value`.

- Plumbing: Ensure these parameters are correctly passed down through `apply_transform` to the functional `warp_affine`.

- Verification: Verify that the functional `warp_affine` correctly handles the fill mode with custom values (especially for masks/integers) and implement the masking logic if it isn't fully supported yet.

- Docs & Tests: Add the missing documentation and a regression test using a segmentation mask scenario.
